### PR TITLE
feat(frontend): avoid username collision with group mentions

### PR DESCRIPTION
**Pull request by grignards**
#3327 

## Checkpoints

These points must be checked before merging. Please don't edit them out.

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

File: frontend_lib/src/helper.js [hunk 1]
```diff
@@ -390,7 +388,73 @@ export const serialize = (objectToSerialize, propertyMap) => {
...
+      isUsernameValid: false,
+      usernameInvalidMsg: props.t('Allowed characters: {{allowedCharactersUsername}}', { allowedCharactersUsername: ALLOWED_CHARACTERS_USERNAME })
+    }
+  }
+
+  const fetchReservedUsernames = await getReservedUsernames(apiUrl)
+  if (fetchReservedUsernames.status !== 200 || !fetchReservedUsernames.json.items) {
+    throw new Error(props.t('Error while checking reserved usernames'))
+  }
+  if (fetchReservedUsernames.json.items.indexOf(username) >= 0) {
```
Comment (gbonaspetti) [inline frontend_lib/src/helper.js:438]:
```suggestion
  if (fetchReservedUsernames.json.items.indexOf(username) >== 0) {
```

File: frontend_lib/src/action.async.js [hunk 2]
```diff
@@ -68,3 +68,35 @@ export const putContentRestoreDelete = (apiUrl, workspaceId, contentId) => {
...
   })
 }
+
+export const getUsernameAvailability = async (apiUrl, username) => {
+  const response = await fetch(`${apiUrl}/system/username-availability?username=${username}`,
+    {
+      credentials: 'include',
+      headers: {
+        ...FETCH_CONFIG.headers
+      },
```
Comment (gbonaspetti) [inline frontend_lib/src/action.async.js:78]:
```suggestion
      headers: FETCH_CONFIG.headers,
```

This comment is just a suggestion, if you prefer stay as it is, you can resolve the conversation.

**Review by gbonaspetti — CHANGES_REQUESTED** [review]

File: frontend_lib/src/helper.js (hunk 1 above)
Comment (grignards) [reply frontend_lib/src/helper.js:438]: ? I do not think this operator exists (at least firefox and node.js give me a syntax error if I use it in a test).

**Review by gbonaspetti — APPROVED** [review]

**Comment by metzgegu** [comment]
Tested on branch

**Pull request merged**